### DISCUSSION TITLE
[Backend] New Codegen for TAPA/AutoBridge

### DISCUSSION THREAD

**Discussion by chhzh123** [cornell-zhang/allo#210] [Projects]
Integrating [AutoBridge/TAPA](https://github.com/rapidstream-org/rapidstream-tapa) is necessary for scaling up the dataflow designs. @EthanMeng324 is leading this project.

**Comment by HanMeng-Ethan** [comment]
Progress for last 3 weeks:
1. Task-based kernel invoking for TAPA codegen
2. A docker image for running TAPA software simulation (can't run directly on server)
3. TAPA stream classifying (input/output) in Allo dataflow, and corresponding codegen
4. Unused stream argument detection and removal in the kernel, conforming to TAPA input output checking
5. New draining strategy for tiled systolic array

Currently the codegen can generate kernel code run directly with customized host code for software simulation, and generate the result correctly.

Future work:
1. The buffer strategy in the vitis codegen can't work in TAPA, probably will use stream for better performance and correctness for the 2D array (might end up with different MLIR from vitis)
2. Need to start trying hardware simulation and hardware test for TAPA
3. The codegen for host code and makefile still need to be done
